core/templates/pages/library-page/search-selectors/subject-selector.component.spec.ts

```diff
@@ -0,0 +1,252 @@
+        },
+        NgbModal,
+      ],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    selectionDetailsStub = {
+      categories: {
+        description: 'description',
+        itemsName: 'categories',
+        masterList: [
+          {
+            id: 'id',
+            text: 'category 1',
+          },
+          {
+            id: 'id_2',
+            text: 'category 2',
+          },
+          {
+            id: 'id_3',
+            text: 'category 3',
+          },
+        ],
+        selections: {id: true, id_2: true, id_3: true},
+        numSelections: 0,
+        summary: 'all categories',
+      },
+      languageCodes: {
+        description: 'English',
+        itemsName: 'languages',
+        masterList: [
+          {
+            id: 'en',
+            text: 'English',
+          },
+          {
+            id: 'es',
+            text: 'Spanish',
+          },
+        ],
+        numSelections: 1,
+        selections: {en: true},
+        summary: 'English',
+      },
+    };
+
+    fixture = TestBed.createComponent(SubjectSelectorComponent);
+    component = fixture.componentInstance;
+    i18nLanguageCodeService = TestBed.inject(I18nLanguageCodeService);
+    spyOnProperty(
+      i18nLanguageCodeService,
+      'onPreferredLanguageCodesLoaded'
+    ).and.returnValue(preferredLanguageCodesLoadedEmitter);
+    translateService = TestBed.inject(TranslateService);
+    searchService = TestBed.inject(SearchService);
+    modalService = TestBed.inject(NgbModal);
+    languageUtilService = TestBed.inject(LanguageUtilService);
+    constructTranslationIdsService = TestBed.inject(
+      ConstructTranslationIdsService
+    );
+    windowDimensionsService = TestBed.inject(WindowDimensionsService);
+    component.ngOnInit();
+    fixture.detectChanges();
+  });
+
+  it('should determine if mobile view is active', () => {
+    const windowWidthSpy = spyOn(
+      windowDimensionsService,
+      'getWidth'
+    ).and.returnValue(766);
+    expect(component.isMobileViewActive()).toBe(true);
+    windowWidthSpy.and.returnValue(1000);
+    expect(component.isMobileViewActive()).toBe(false);
+  });
+
+  it('should update selection details if there are no selections', () => {
+    spyOn(translateService, 'instant').and.returnValue('key');
+    component.updateSelectionDetails('categories');
+    let selectionDetails = component.selectionDetails;
+    expect(selectionDetails.categories.numSelections).toEqual(0);
+  });
+
+  it('should update selection details', () => {
+    searchService.selectionDetails = selectionDetailsStub;
+    component.updateSelectionDetails('categories');
+    expect(component.selectionDetails).toEqual(selectionDetailsStub);
+  });
+
+  it('should initialize', () => {
+    spyOn(component, 'updateSelectionDetails');
+    component.ngOnInit();
+    expect(component.updateSelectionDetails).toHaveBeenCalled();
+  });
+
+  it('should detect selections', () => {
```

Comment:
detect selections of what? please be more specific.

Reply:
done